catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DisplayLocation.scala

```diff
@@ -2,21 +2,43 @@ package uk.ac.wellcome.platform.api.models
 
 import com.fasterxml.jackson.annotation.JsonProperty
 import io.swagger.annotations.{ApiModel, ApiModelProperty}
-import uk.ac.wellcome.models.Location
+import uk.ac.wellcome.models.{DigitalLocation, Location, PhysicalLocation}
 
 @ApiModel(
   value = "Location",
-  description = "A location that provides access to an item"
+  description = "A location that provides access to an item",
+  subTypes =
+    Array(classOf[DisplayDigitalLocation], classOf[DisplayPhysicalLocation])
 )
-case class DisplayLocation(
+sealed trait DisplayLocation
+
+object DisplayLocation {
+  def apply(location: Location): DisplayLocation = location match {
+    case l: DigitalLocation =>
+      DisplayDigitalLocation(
+        locationType = l.locationType,
+        url = l.url,
+        credit = l.credit,
+        license = DisplayLicense(l.license)
+      )
+    case l: PhysicalLocation =>
+      DisplayPhysicalLocation(locationType = l.locationType, label = l.label)
+  }
+}
+
+@ApiModel(
+  value = "DigitalLocation",
+  description = "A digital location that provides access to an item"
+)
+case class DisplayDigitalLocation(
   @ApiModelProperty(
     value = "The type of location that an item is accessible from.",
     allowableValues = "thumbnail-image, iiif-image"
   ) locationType: String,
   @ApiModelProperty(
     dataType = "String",
     value = "The title or other short name of a license"
-  ) url: Option[String] = None,
+  ) url: String,
```

Comment:
Why have you changed this to be non-optional? Our usual policy is to omit strings that are empty – e.g. description, lettering.

(And should we change the description here?)

Reply:
It doesn't seem to be possible for it to be None at least checking ho the transformer fills it in. Especially with it being a `DigitalLocation`, `url` must exist

(Yep, the description is wrong)